### PR TITLE
core: disallow multiple write stmts in single connection and poison tx if half-done write stmt is abandoned

### PR DESCRIPTION
**Pull request by jussisaurio**
Took Claude Fable for a test drive just for writing this PR description, and it actually turned out pretty good. AGI confirmed.

---

## What this PR does

Two things, in two commits:

1. **A connection can now run only one write statement at a time.** Starting a second write statement while another is mid-flight returns `SQLITE_BUSY`. Additionally, if a half-finished write statement inside a `BEGIN ... COMMIT` transaction is abandoned, the transaction refuses to commit.
2. **Documents a known durability edge case in MVCC**, with tests that document the current behavior so any eventual fix changes them deliberately rather than accidentally.

## Background: why Turso has a problem SQLite doesn't

In SQLite, `sqlite3_step()` never returns control to the application in the middle of modifying the database. A write statement either runs to completion or stops at a clean boundary, like returning a row.

Turso supports asynchronous I/O, so a statement *can* pause in the middle of a write — say, halfway through an `UPDATE` that touches many rows — while waiting for the disk. That's normally fine: the application steps the statement again and it resumes where it left off.

The trouble starts when the application doesn't resume it. If the half-done statement is reset or dropped, or another write statement starts in the meantime, the transaction now contains changes from a statement that never finished, and we have to decide what's safe to commit and what must be undone.

## Commit 1: one writer at a time, and poisoned transactions

**One writer at a time.** If a second write statement could start while the first is suspended mid-write, then resetting or dropping either one would require undoing only *that* statement's changes. Our undo mechanisms (statement savepoints) can't cleanly undo that statement's changes due to being a LIFO stack. Rather than risk committing or rolling back the wrong thing, the second write statement now gets `SQLITE_BUSY`: finish or reset the active one first. Reads can still run alongside a write.

This is a deliberate divergence from SQLite, which allows overlapping write statements — it can, because it never pauses mid-mutation. Documented in COMPAT.md.

**Poisoning.** If a write statement inside an interactive tx is reset or dropped before finishing, and no statement savepoint exists that could undo just that statement, the transaction is marked rollback-only ("poisoned"). A later `COMMIT` rolls everything back and returns an error instead of committing a half-applied statement. `ROLLBACK` clears the marker as usual.

## Commit 2: the MVCC durability gap (documented, not yet fixed)

In MVCC, all concurrent statements on one connection share a single MVCC transaction. A write statement that finishes while a sibling statement (say, an open `SELECT` being stepped row by row) is still active cannot commit yet, because committing would end the transaction the `SELECT` is still reading from. So the commit is deferred until the last statement finishes.

The gap: between "your `INSERT` returned success" and "the last sibling statement finished", the data is not durable. Worse, if the shared transaction ends in a rollback — the open `SELECT` is dropped mid-scan, or a later write statement fails after changing rows — the `INSERT` that already reported success is silently rolled back.

SQLite never has this window: it commits when the writer itself finishes and lets the remaining statements carry on under a read-only transaction. The proper fix is the MVCC equivalent — commit at the writer's completion and move remaining readers onto a fresh read-only transaction that sees exactly the same data. A FIXME at the deferral site in `core/vdbe/execute.rs` spells this out. Until that lands, this commit documents the gap in COMPAT.md and pins both loss scenarios with tests.

### CODE REVIEW

File: core/vdbe/execute.rs [hunk 1]
```diff
@@ -3464,22 +3483,50 @@ pub fn op_transaction_inner(
...
+                // does not return to the caller in the middle of built-in
+                // write opcodes. Turso can suspend there for async I/O, so a
+                // second writer would make reset/drop cleanup hard to get right.
+                if statement_writes_db
+                    && !conn.is_nested_stmt()
+                    && !state.is_active_write
+                    && active_writers > 0
+                {
+                    return Err(LimboError::Busy);
+                }
```
Comment (pereman2) [inline core/vdbe/execute.rs:3555]: This seems like could cause a TOCTOU bug. One statement could read `active_writers == 0`, then on `CheckSchemaCookie` it could return I/O then another statement gets to same place reads `active_writers == 0` and both of them succeed to start.

**Comment by LeMikaelF** [comment]
Closes https://github.com/tursodatabase/turso/issues/7345

File: core/vdbe/execute.rs (hunk 1 above)
Comment (jussisaurio) [reply core/vdbe/execute.rs:3555]: do we care about TOCTOU wrt multiple statements & single connection? sharing connection across threads and performing parallel statement executions with that single connection should at least be heavily discouraged as an antipattern...
